src/Tutor.Core/InstructorModel/Instructors/DefaultInstructor.cs

```diff
@@ -1,17 +1,93 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using FluentResults;
 using Tutor.Core.DomainModel.AssessmentEvents;
+using Tutor.Core.DomainModel.KnowledgeComponents;
 
 namespace Tutor.Core.InstructorModel.Instructors
 {
     public class DefaultInstructor : IInstructor
     {
-        public DefaultInstructor()
+        private readonly IKCRepository _ikcRepository;
+        private readonly IAssessmentEventRepository _assessmentEventRepository;
+
+        public DefaultInstructor(IKCRepository ikcRepository, IAssessmentEventRepository assessmentEventRepository)
         {
+            _ikcRepository = ikcRepository;
+            _assessmentEventRepository = assessmentEventRepository;
         }
 
         public Result<AssessmentEvent> SelectSuitableAssessmentEvent(int knowledgeComponentId, int learnerId)
         {
-            throw new System.NotImplementedException();
+            if (_ikcRepository.GetAssessmentEventsByKnowledgeComponent(knowledgeComponentId).Count == 0) return Result.Fail("There are no assessment events yet.");
```

Comment:
We have an AERepository and the KCRepository, where it seems that both repositories can return AEs. We should either merge both repositories or more clearly define what can be expected from which repo (and choose suitable names). Which option do you think is better and why?